middleware/locale/locale.go

```diff
@@ -0,0 +1,56 @@
+package locale
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/mholt/caddy/middleware"
+	"github.com/mholt/caddy/middleware/locale/method"
+)
+
+// Locale is a middleware to detect the user's locale.
+type Locale struct {
+	Next     middleware.Handler
+	Locales  []string
+	Methods  []method.Method
+	Settings *method.Settings
+}
+
+// ServeHTTP implements the middleware.Handler interface.
+func (l *Locale) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
```

Comment:
This is not scoped? I'd expect it to be. Why would I want language detection, for example, with *proxy* or *upload*.